blockmanager.go

```diff
@@ -662,76 +662,63 @@ func (b *blockManager) handleDonePeerMsg(peer *peerpkg.Peer) {
 // given error. This function can convert some select blockchain and mempool
 // error types to the historical rejection codes used on the p2p wire protocol.
 func errToWireRejectCode(err error) (wire.RejectCode, string) {
-	// Unwrap mempool errors.
-	var rerr mempool.RuleError
-	if errors.As(err, &rerr) {
-		err = rerr.Err
-	}
-
 	// The default reason to reject a transaction/block is due to it being
 	// invalid somehow.
 	code := wire.RejectInvalid
 	var reason string
 
-	var berr blockchain.RuleError
-	var terr mempool.TxRuleError
+	// Convert recognized errors to a reject code.
 	switch {
-	case errors.As(err, &berr):
-		// Convert the chain error to a reject code.
-		switch berr.ErrorCode {
-		// Rejected due to duplicate.
-		case blockchain.ErrDuplicateBlock:
-			code = wire.RejectDuplicate
+	// Rejected due to duplicate.
+	case errors.Is(err, blockchain.ErrDuplicateBlock):
+		code = wire.RejectDuplicate
+		reason = err.Error()
 
-		// Rejected due to obsolete version.
-		case blockchain.ErrBlockVersionTooOld:
-			code = wire.RejectObsolete
+	// Rejected due to obsolete version.
+	case errors.Is(err, blockchain.ErrBlockVersionTooOld):
+		code = wire.RejectObsolete
+		reason = err.Error()
 
 		// Rejected due to checkpoint.
```

Comment:
Looks to be indented one too many.  Not gofmt?

Reply:
using `goimports`, linter checks with `gofmts` however.